### PR TITLE
tweak(ui): support `sameWeek` format

### PR DESCRIPTION
**Pull request by shhdgit**
close: #936

### CODE REVIEW

**Comment by shhdgit** [comment]
/cc @breeswish @unbyte 
PTAL.

**Review by breeswish — APPROVED** [review]
This is an interesting case showing that how hard localization is.. Different language properties affects calculation logic!

File: ui/lib/components/DateTime/calendarPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        : diff < 0
+        ? 'lastDay'
+        : diff < 1
+        ? 'sameDay'
+        : diff < 2
+        ? 'nextDay'
+        : weekDiff === 0
+        ? 'sameWeek'
+        : 'nextWeek'
+    debugger
```
Comment (unbyte) [inline ui/lib/components/DateTime/calendarPlugin.ts:61]: oh no

File: ui/lib/components/DateTime/Calendar.tsx [hunk 2]
```diff
@@ -7,23 +7,27 @@ import i18next from 'i18next'
...
     nextDay: '[Tomorrow] h:mm A',
-    nextWeek: 'dddd h:mm A',
+    nextWeek: '[Next] dddd h:mm A',
     lastDay: '[Yesterday] h:mm A',
     lastWeek: '[Last] dddd h:mm A',
     sameElse: 'lll',
   },
   zh: {
     sameDay: '[今天] HH:mm',
+    sameWeek: 'dddd HH:mm',
```
Comment (unbyte) [inline ui/lib/components/DateTime/Calendar.tsx:30]: how about '[本]dddd HH:mm'

File: ui/lib/components/DateTime/calendarPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export default (o, c, d) => {
+  const LT = 'h:mm A'
+  const L = 'MM/DD/YYYY'
+  const calendarFormat = {
+    lastDay: `[Yesterday at] ${LT}`,
+    sameDay: `[Today at] ${LT}`,
+    nextDay: `[Tomorrow at] ${LT}`,
+    sameWeek: `dddd [at] ${LT}`,
```
Comment (unbyte) [inline ui/lib/components/DateTime/calendarPlugin.ts:32]: how about '[This]dddd [at] ${LT}'

**Review by unbyte — COMMENTED** [review]
LGTM

File: ui/lib/components/DateTime/calendarPlugin.ts (hunk 1 above)
Comment (shhdgit) [reply ui/lib/components/DateTime/calendarPlugin.ts:61]: Oh, no. This is a tragedy.

**Comment by unbyte** [comment]
> Different language properties affects calculation logic!

And different cultures have different definitions of the beginning day of a week...

File: ui/lib/components/DateTime/calendarPlugin.ts (hunk 1 above)
Comment (shhdgit) [reply ui/lib/components/DateTime/calendarPlugin.ts]: Done, Thanks!

File: ui/lib/components/DateTime/Calendar.tsx (hunk 2 above)
Comment (shhdgit) [reply ui/lib/components/DateTime/Calendar.tsx:30]: I think `星期x 18:00` is clear enough. And `本周x 18:00` looks good.

**Pull request merged**